source/material_model/rheology/strain_dependent.cc

```diff
@@ -283,6 +305,32 @@ namespace aspect
         friction_strain_weakening_factors = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Friction strain weakening factors"))),
                                                                                     n_fields,
                                                                                     "Friction strain weakening factors");
+
+        if (prm.get ("Strain healing mechanism") == "no healing")
+          healing_mechanism = no_healing;
+        else if (prm.get ("Strain healing mechanism") == "temperature dependent")
+          healing_mechanism = temperature_dependent;
+        else
+          AssertThrow(false, ExcMessage("Not a valid Strain healing mechanism!"));
+
+        // Currently this functionality only works in field composition
+        if (healing_mechanism != no_healing && this->introspection().compositional_name_exists("particles"))
```

Comment:
This line is still a problem. I think the check we want is whether the list of particle properties includes `viscoplastic strain invariants`, rather than `this->introspection().compositional_name_exists("particles")`.

@gassmoeller - Is there a way straightforward way to access the list of particle properties here?

Reply:
Long-term, we can avoid this by having the viscoelastic strain invariants particle property obtain values from the rheology module, rather than recalculating the values.

Reply:
I think this should work approximately as follows:
```
if (healing_mechanism != no_healing)
{
    const Postprocess::Particles<dim> &particle_postprocessor =
      postprocess_manager.template get_matching_postprocessor<Postprocess::Particles<dim> >();
    const Particle::Property::Manager<dim> *particle_property_manager = &particle_postprocessor.get_particle_world().get_property_manager();

             AssertThrow(particle_property_manager.plugin_name_exists("viscoplastic strain invariants") == false, ExcMessage("This healing mechanism currently does not work if the strain is tracked on particles."));
}
```
For this to work you will need to at least add the following headers to the top of the file:
```
#include <aspect/postprocess/particles.h>
#include <aspect/particle/property/interface.h>
```
and maybe also `aspect/simulator.h` (not sure at the moment, try it if it does not compile without).